src/errors.rs

```diff
@@ -73,6 +93,15 @@ macro_rules! verify_err {
     }};
 }
 
+macro_rules! arg_err {
```

Comment:
Functions are nicer than macros. I would remove this macro and rewrite it as a function:
```
impl InternalError {
    fn invalid_argument(error_message: &'static str) -> Self {
        InternalError::InvalidArgument(String::from(error_message))
    }
}
```

Reply:
I styled this one after `verify_err!`, so whatever we do to one we should do to the other. Should I go ahead and swap both or save this for later?

Reply:
Right, I saw there was other macros used in the code. I agree we should do it all in one go. We should save it for later then.

I will make a ticket for it, but it is not high priority.

Reply:
This is now tracked on #152